src/java_bytecode/java_entry_point.cpp

```diff
@@ -123,13 +170,19 @@ bool java_static_lifetime_init(
       code_function_callt function_call;
       function_call.lhs()=nil_exprt();
       function_call.function()=it->second.symbol_expr();
+      function_call.add_source_location()=source_location;
       code_block.add(function_call);
     }
   }
 
   return false;
 }
 
+static bool is_string_array(const typet& t)
```

Comment:
There are many functions like this (is_constructor? is_assertion_error? etc). Can we put them into some common place?